x-pack/plugins/alerting/server/lib/retry_if_not_found.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// This module provides a helper to perform retries on a function if the
+// function ends up throwing a SavedObject 404 Not Found. This can happen
+// when the various socket hang up, ECONNRESET or some "normal" lifecycle
+// events are occurring: migrations, ES | Kibana restarted, etc.
+
+import { Logger, SavedObjectsErrorHelpers } from '../../../../../src/core/server';
+
+type RetryableForNotFound<T> = () => Promise<T>;
+
+// number of times to retry when not found error occur
+export const RetryForNotFoundAttempts = 5;
+
+// milliseconds to wait before retrying when not found error occur
+// note: we considered making this random, to help avoid a stampede, but
+// with 1 retry it probably doesn't matter, and adding randomness could
+// make it harder to diagnose issues
+const RetryForNotFoundDelay = 600;
```

Comment:
After looking at some existing cases, I've seen scenarios where transient 404 errors can happen for longer than 15 seconds. However, this PR seems to configure the retry logic to stop after 3 seconds (5 attempts w/ 600ms delay). What number would we be comfortable retrying, given we don't know when transient errors stop?

In Cloud's use case, from what I understand, a new proxy service starting up can cause these. Do we know how long it takes for the proxy to start up?